libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -754,6 +837,342 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
+        willInfo.qos = ( MQTTQoS_t ) pConnectInfo->pWillInfo->qos;
+
+        /* Getting Connect packet size using MQTT V4_beta2 API*/
+
+        mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                                &willInfo,
+                                                &remainingLength,
+                                                pPacketSize );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+
+        /*Allocating memory for Connect packet*/
+        networkBuffer.pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.size = *pPacketSize;
+
+        /*Serializing the connect packet using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_SerializeConnect( &connectInfo,
+                                            &willInfo,
+                                            remainingLength,
+                                            &( networkBuffer ) );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+    }
+    else
+    {
+        /* Getting Connect packet size using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                                NULL,
+                                                &remainingLength,
+                                                pPacketSize );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+
+        /*Allocating memory for Connect packet*/
+        networkBuffer.pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.size = *pPacketSize;
+
+        /*Serializing the connect packet using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_SerializeConnect( &connectInfo,
+                                            NULL,
+                                            remainingLength,
+                                            &( networkBuffer ) );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+    }
+
+    *pConnectPacket = networkBuffer.pBuffer;
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+/* Disonnect Serialize Wrapper*/
+static IotMqttError_t _disconnectSerializeWrapper( uint8_t ** pDisconnectPacket,
+                                                   size_t * pPacketSize )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    MQTTFixedBuffer_t networkBuffer;
+
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+
+    /* Getting Disconnect packet size using MQTT V4_beta2 API*/
+    mqttStatus = MQTT_GetDisconnectPacketSize( pPacketSize );
+
+    /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+    status = convertReturnCode( mqttStatus );
```

Comment:
I don't think we need to proceed if the MQTT_GetDisconnectPacketSize call returns an error.